state-synchronizer/src/executor_proxy.rs

```diff
@@ -67,21 +70,23 @@ pub trait ExecutorProxyTrait: Sync + Send {
 }
 
 pub(crate) struct ExecutorProxy {
-    storage_read_client: Arc<StorageReadServiceClient>,
+    storage_read_client: Arc<dyn StorageRead>,
     executor: Arc<Mutex<Executor<LibraVM>>>,
     reconfig_subscriptions: Vec<ReconfigSubscription>,
-    on_chain_configs: Arc<HashMap<ConfigID, Vec<u8>>>,
+    on_chain_configs: OnChainConfigPayload,
 }
 
 impl ExecutorProxy {
-    pub(crate) fn new(
+    pub(crate) async fn new(
         executor: Arc<Mutex<Executor<LibraVM>>>,
         config: &NodeConfig,
         reconfig_subscriptions: Vec<ReconfigSubscription>,
     ) -> Self {
         let storage_read_client = Arc::new(StorageReadServiceClient::new(&config.storage.address));
 
-        let on_chain_configs = Arc::new(HashMap::new());
+        let on_chain_configs = Self::fetch_all_configs(storage_read_client.clone())
```

Comment:
should we update state sync to not use the old storage client?

Reply:
I was hoping aaron would just convert all

Reply:
No way to convince you to clean this up so we don't end up spreading async?

Reply:
probably not in this pr 😆